src/components/FilterGroup.tsx

```diff
@@ -85,7 +91,7 @@ export function FilterGroup({
       {renderTitle()}
       <CollapsibleSection className={cssClasses.optionsContainer}>
         {searchable && <SearchInput className={cssClasses.searchInput} />}
-        {filterOptions.map(o => {
+        {filterOptions.slice(0, showAll ? filterOptions.length : showMoreLimit).map(o => {
```

Comment:
did you test if this works properly when typing in the search input? it seems like the search would only be performed on this slice of filter options instead of searching all of the options and then displaying the first few